### PR TITLE
give xdmod-reader SA additional permission to query namespaces

### PR DESCRIPTION
**Pull request by naved001**
This PR creates a clusterrole and a binding to do that. This also unbinds the
SA from the other clusterrolebinding since those rules are covered under the
new clusterole.

### CODE REVIEW

File: cluster-scope/overlays/ocp-staging/clusterroles/xdmod-permissions/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: xdmod-permissions
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  verbs:
+  - get
+  - list
```
Comment (naved001) [inline cluster-scope/overlays/ocp-staging/clusterroles/xdmod-permissions/clusterrole.yaml:12]: `list` is the additional permission that was needed. the pervious rolebinding only had `get`

**Review by larsks — APPROVED** [review]
Looks good, although if you're feeling motivated I would suggest updating the commit message to include a little more detail (e.g., including something like "...a service account needs both `get` and `list` verbs on namespaces in order to access metrics").

**Comment by larsks** [comment]
> it actually doesn't. You can fetch metrics just fine with get, but @tzumainn https://github.com/CCI-MOC/ops-issues/issues/625#issuecomment-1188471696 to query namespaces which is why I added list in this PR.

That seems worth including in the commit message (it answers the question, "why did we make this change?")

**Comment by naved001** [comment]
@larsks oops, you commented right after I merged this. sorry.

**Pull request merged**